### ISSUE TITLE
Refresh movie task gets stuck

### ISSUE THREAD

**Issue by user2223** [Radarr/Radarr#4897]
**Describe the bug**
Refresh Movie Task gets stuck after a few movies every single time without any strange output in tracert log. 
I tried to reinstall with a new database -> same problem after the upgrade to v3. The refresh task works fine with radarr v2.
I tried to get help on discord -> no responses, it seems like a problem nobody else has.
The only answer i got was, that external scrips may cause radarr to pause and wait for an response from the script. But i don't use any external scripts. Especially not after the reinstall.

1. Start Radarr.
2. Wait for refresh movie task to start.
3. Wait for it to stop after a few movies.
4. Nothing happens. Task is running for days if nobody closes radarr.

**Expected behavior**
Movies get refreshed. Task ends. 

**Platform Information (please complete the following information):**
 - OS: Windows Server 2016
 - Radarr Version 3.0.0.3445 but the problem started happening a few months ago so the bug is not really new. 

**Debug Logs**
https://pastebin.com/WXPnnmqi 
2020-8-17 08:18:09 is the timestamp for the last movie that got refreshed successfully.
I can give you a few of these but i can't find anything useful.

Please tell me what you need from me to fix this problem.

**Comment by user2223** [comment]
> 
> 
> This occurred to me as well and I spent one too many hours debugging the issue but unfortunately I was unable to pin down the bug, however, I was able to resolve/fix the issue and de-stuck the refresh task.
> 
>     1. Bump up the logging to `debug` to make your life easier
> 
>     2. Run the refresh task
> 
>     3. Once the task fails/restarts open up the log and find the last ~5 videos before the task failed
> 
>     4. Simply remove those ~5 movies and place them into a temporary directory
> 
>     5. With any luck Radarr will be able to complete a full refresh task
> 
>     6. Once a full refresh task has completed remove all the meta data within the video directories and simply move them back into your download/pipeline directory so Radarr re-imports
> 
> 
> If you could post the contents/permissions of the videos so we can share and compare to hopefully find a common denominator.

Sadly it happens for random movies. I just removed the first 100 movies because it just kept hanging on different movies everytime. Radarr just stops sometime after 10 to 50 movies. Sometimes it gets further and sometimes it stops after a few movies. I can't identify a single movie that continuously causes the hang. But i will test further and see if i can find something.





> 
> 
> Mediainfo hang up possibly.. Can't recall if that is called in a refresh or only import

I guess you mean the "Analyse video files" setting? I tried to disable it but nothing changed. It also says |Debug|UpdateMediaInfoService|MediaInfo is disabled for every movie.

**Comment by user2223** [comment]
It seems like one of the last updates fixed the Issue. I just updated from version 3.0.0.3494 to 3.0.0.3502 and radarr just finished the refresh task without any issue. I will report back if it starts to happen again.

**Comment by user2223** [comment]
Okay it seems like it was just luck or the problem is back.
Same problem like before. Still no explanation in the logs. I really don't know what could cause this. Here is my most recent debug log. 
Thanks for any assistance to find the cause of this.
https://pastebin.com/qSJZZMST

**Comment by ta264** [comment]
Can you post trace logs please

**Comment by user2223** [comment]
https://pastebin.com/sUpwD57V There you go. I switched to debug after artisin's comment.

**Comment by artisin** [comment]
Welp, the mystery continues and I've been afraid to update for this exact reason. I'm assuming you've already tried to revert to 3.0.0.349, remove the movies,  and upgrade from 3.0.0.3494 to 3.0.0.3502  again?

**Comment by user2223** [comment]
> 
> 
> Welp, the mystery continues and I've been afraid to update for this exact reason. I'm assuming you've already tried to revert to 3.0.0.349, remove the movies, and upgrade from 3.0.0.3494 to 3.0.0.3502 again?

I tried to downgrade to 3.0.0.3502 but that didn't help (I didn't use the backed up database though). The removal of movies never made any difference. It seems like every movie might get it stuck. 
I will try to downgrade to 3.0.0.349, use the proper backup files and upgrade to 3.0.0.3502 again to see if it changes anything. Probably not before sunday though because I'm not at home right now.

**Comment by bakerboy448** [comment]
> https://pastebin.com/sUpwD57V There you go. I switched to debug after artisin's comment.

Looks like it is stuck and stops after this line
2020-9-3 13:17:13.0|Info|RefreshMovieService|Updating info for Air Force One

**Comment by user2223** [comment]
Now there is just more broken stuff. Adding movies breaks after a few (immediatly if i bulk add) and the list sync gets stuck too. I feel like it's a problem with my machine rather than radarr. I will setup a completely new instance on a fresh vm and see if that fixes my problems. Thank you all for your help even if we didn't really find a solution. 
I will report back if the problem occurs on a totally different machine too.

**Comment by user2223** [comment]
Okay, now I'm out of ideas. I setup a totally new ubuntu 20.04 VM. I installed radarr V2 and mounted the volumes by cifs. I followed this instructions for the setup :https://github.com/Radarr/Radarr/wiki/Installation.
I used this v2 instance to build a completley new database from the existing movie folders. Everything worked fine. I used this installation for about 1 month now and I didn't have any issues. Everything worked as expected.

Before i made the Update to V3 I created a full backup and a snapshot, just in case. The update to v3 worked fine, just some confusion about changing away from mono but it worked.
Well, now it doesn't. It's exactly the same behavior like before. The refresh task starts and gets stuck after a few movies. Sometimes earlier, sometimes later but it doesn't finish. 
I'm 100% sure that it worked just fine for the first couple of refreshes because I checked immediately.

I'm seriously out of ideas and don't know what to do next. I might just stick to v2.

This is the tracert log: https://pastebin.com/mAG4eTUB

**Comment by bakerboy448** [comment]
Is this still occuring on the more recent v3 versions?

**Comment by user2223** [comment]
Yes it's still happening. Here is a recent trace log: https://pastebin.com/st1GxsTD
I will try to run the mono version of v3 next, i feel like it worked before the switch to .NET Core. I know that the refresh task ran successfully at least once (even after the switch to .Net Core) but it failed it's next regular run and i can't gest past the first few movies now.

**Comment by bakerboy448** [comment]
@user2223 reopened

that is a debug, not a trace however.

**Comment by user2223** [comment]
Sorry about that. Here you go: https://pastebin.com/LGVem47H it's a little longer now

**Comment by bakerboy448** [comment]
it seems to just die after `10 Days in a Madhouse`

But it may just be the download client noise is making it hard to follow?

Not to be a pain, but a clean log with an empty (or disconnected) Activity queue and just the refresh running would be clutch.

But it does appear to be getting stuck, the question is on what and why

what movie is after 10 days? (alphabetically)

**Comment by user2223** [comment]
I created a log without indexers or download clients enabled. 

> Not to be a pain, but a clean log with an empty (or disconnected) Activity queue and just the refresh running would be clutch.

You are not a pain, I'm really grateful for your help and will be extremly happy if we find the cause for this problem. 

Sadly it doesn't stop at the same movies every time, sometimes it gets further sometimes it stops after 10 movies. The moview after 10 Days in a Madhouse would be 10 Minutes Gone. But it refreshed that movie fine just now.

In the case of my new log it got until Aliens (2020-11-16 19:47:06.0|Info|RefreshMovieService|Updating info for Aliens).

He got two covers and got stuck after that:

> 2020-11-16 19:47:06.1|Info|MediaCoverService|Downloading Poster for [Alien Resurrection (1997)][tt0118583, 8078] https://image.tmdb.org/t/p/original/kzslFL1n01WI5lEcYzFSUi3Quik.jpg

> 2020-11-16 19:47:06.3|Info|MediaCoverService|Downloading Fanart for [Alien Resurrection (1997)][tt0118583, 8078] https://image.tmdb.org/t/p/original/f8XUY6srsxTabJ8h353izrqaKfs.jpg


There is an error at 2020-11-16 19:51:00.2|Trace|SignalRMiddleware| which i didn't see before maybe that has some connection (it happened about 4 minutes after getting stuck though. 

Here is the new cleaner trace log: https://paste.ee/p/Ej7de

**Comment by bakerboy448** [comment]
```
2020-11-16 19:50:59.4|Trace|CommandQueueManager|Checking if command is queued or started: RefreshMovie
2020-11-16 19:50:59.4|Trace|CommandQueueManager|Command is already in progress: RefreshMovie
2020-11-16 19:51:00.1|Trace|Http|Req: 191 [GET] /logfile/radarr.trace.0.txt (from ::ffff:192.168.0.235 Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:82.0) Gecko/20100101 Firefox/82.0)
2020-11-16 19:51:00.1|Trace|Http|Res: 191 [GET] /logfile/radarr.trace.0.txt: 200.OK (1 ms)
2020-11-16 19:51:00.2|Trace|SignalRMiddleware|System.OperationCanceledException: The operation was canceled.
   at System.Threading.CancellationToken.ThrowOperationCanceledException()
   at Nancy.NancyEngine.HandleRequest(Request request, Func`2 preRequest, CancellationToken cancellationToken)
   at Nancy.Owin.NancyMiddleware.<>c__DisplayClass2_1.<<UseNancy>b__1>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Owin.WebSocketAcceptAdapter.<>c__DisplayClass6_0.<<AdaptWebSockets>b__0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at Radarr.Host.Middleware.SignalRMiddleware.<Attach>b__7_0(HttpContext context, Func`1 next) in D:\a\1\s\src\NzbDrone.Host\WebHost\Middleware\SignalRMiddleware.cs:line 58

[v3.0.0.4079] System.OperationCanceledException: The operation was canceled.
   at System.Threading.CancellationToken.ThrowOperationCanceledException()
   at Nancy.NancyEngine.HandleRequest(Request request, Func`2 preRequest, CancellationToken cancellationToken)
   at Nancy.Owin.NancyMiddleware.<>c__DisplayClass2_1.<<UseNancy>b__1>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Owin.WebSocketAcceptAdapter.<>c__DisplayClass6_0.<<AdaptWebSockets>b__0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at Radarr.Host.Middleware.SignalRMiddleware.<Attach>b__7_0(HttpContext context, Func`1 next) in D:\a\1\s\src\NzbDrone.Host\WebHost\Middleware\SignalRMiddleware.cs:line 58
2020-11-16 19:51:29.4|Trace|Scheduler|Pending Tasks: 2
```



^unrelated?, but seems slightly odd?
and, appears this is where it died

```
2020-11-16 19:47:06.3|Debug|ManagedHttpDispatcher|Downloading [https://image.tmdb.org/t/p/original/f8XUY6srsxTabJ8h353izrqaKfs.jpg] to [/home/felix/.config/Radarr/MediaCover/68/fanart.jpg]
2020-11-16 19:47:06.4|Trace|ConfigService|Using default config value for 'proxyenabled' defaultValue:'False'
2020-11-16 19:47:06.5|Debug|ManagedHttpDispatcher|Downloading Completed. took 0s
2020-11-16 19:47:06.5|Debug|MediaCoverService|Resizing Poster-500 for [Alien Resurrection (1997)][tt0118583, 8078]
2020-11-16 19:47:06.6|Debug|MediaCoverService|Resizing Poster-250 for [Alien Resurrection (1997)][tt0118583, 8078]
2020-11-16 19:47:06.7|Debug|MediaCoverService|Resizing Fanart-360 for [Alien Resurrection (1997)][tt0118583, 8078]
2020-11-16 19:47:06.8|Debug|MediaCoverService|Resizing Fanart-180 for [Alien Resurrection (1997)][tt0118583, 8078]
2020-11-16 19:47:06.9|Trace|EventAggregator|Publishing MediaCoversUpdatedEvent
2020-11-16 19:47:06.9|Trace|EventAggregator|MediaCoversUpdatedEvent -> ExtraService
2020-11-16 19:47:06.9|Debug|CleanExtraFileService|Cleaning missing metadata files for movie: Alien Resurrection
2020-11-16 19:47:06.9|Trace|EventAggregator|MediaCoversUpdatedEvent <- ExtraService
2020-11-16 19:47:06.9|Trace|EventAggregator|MediaCoversUpdatedEvent -> MovieModule
2020-11-16 19:47:06.9|Trace|EventAggregator|MediaCoversUpdatedEvent <- MovieModule
2020-11-16 19:47:06.9|Trace|EventAggregator|MediaCoversUpdatedEvent -> MovieModule
2020-11-16 19:47:06.9|Trace|ConfigService|Using default config value for 'movieinfolanguage' defaultValue:'1'
2020-11-16 19:47:06.9|Trace|EventAggregator|MediaCoversUpdatedEvent <- MovieModule
2020-11-16 19:47:06.9|Trace|EventAggregator|MovieUpdatedEvent <~ MediaCoverService
2020-11-16 19:47:29.3|Trace|Scheduler|Pending Tasks: 2

```

it also looks like you have quite a few unmapped folders in your root folder....you don't have any downloads going to your library/root folder right?

**Comment by user2223** [comment]
> it also looks like you have quite a few unmapped folders in your root folder....you don't have any downloads going to your library/root folder right?
Well it certainly seems like it, i thought i had everything added but it seems like 49 movies are unmapped. I'll get that fixed right away and see if that helps.
No, no downloads going into the library. I rebuild the whole library when i switched to linux and nothing but radarr should have added anything to these folders since then..

**Comment by user2223** [comment]
> Well it certainly seems like it, i thought i had everything added but 49 movies are unmapped. I'll get that fixed right away and see if that helps.

Okay, that didn't help at all. It was a pain because it got stuck very often and i pretty much had to remove most unmapped movies from the folder and simpy move it into another folder and restart the mapping and adding over and over again until it just ran through. 
Step by step:
1.Click Movies -> Import and select root folder with unmapped movies
2.the list loaded fine and sometimes the mapping went pretty smooth. Sometimes it got stuck after a few mappings. Only successful mappings seemed to get it stuck.
3. After a few movies were mapped i clicked cancel processing and tried to add the movies.
4. Sometimes all movies got added successfully, sometimes it didn't add any movies. (It also didn't always successfully download the covers)

I feel like that might be the same reason why the refresh task gets stuck that often. I may try to produce some logs tomorrow but for now I have a clean log without indexers and download client where it got stuck at the very first movie. 

https://pastebin.com/bfrrx4ng

There is an error at 2020-11-18 18:18:39.4|Fatal|RadarrErrorPipeline|Request Failed. GET /MediaCover/3367/poster-250.jpg
The movie 3367 doesn't exist though.

I think it got stuck right here: 2020-11-18 18:13:15.8|Trace|HttpClient|Req: [GET] https://radarrapi.servarr.com/v1/movie/138820

**Comment by user2223** [comment]
Seems to be fixed now. Radarr v3 ist running without issues for almost a week now. I rolled back to my backup and tried the update again and let it run for a few days. Now it just works. I also fixed some ipv6 related network stuff, maybe that helped. Nothing else was changed.

**Comment by farghen** [comment]
This started happening to my install on 2021.03.02.  After removing several movies and restarting I got to one that was unusually large (22gb) and removing that resolved it.  Possibly not enough memory to scan that file on the rig.  Have added other files back and it's still behaving.   

I recall I was also unable to import that one.  I suppose I should re-rip it from disk to a smaller file.

Version
3.0.2.4552
Package Version
3.0.2.4552-ls93 by linuxserver.io
.NET Core
Yes (3.1.10)
Docker
Yes

**Comment by drzoidberg33** [comment]
I'm seeing this behaviour too. Looks like it struggles with a specific file and then starts scanning from the beginning again, the is possibly corrupt but it does play fine so I'm not sure if it's that.

I've just removed the file from Radarr for now and it's no longer getting stuck in this loop.

In the logs I see it just stop abruptly and restart after the problem file:

```
2021-3-16 15:45:22.8|Info|DiskScanService|Scanning disk for Starship Troopers
2021-3-16 15:45:23.2|Info|Bootstrap|Starting Radarr - /app/radarr/bin/Radarr.dll - Version 3.0.2.4552
2021-3-16 15:45:23.4|Info|AppFolderInfo|Data directory is being overridden to [/config]
2021-3-16 15:45:23.5|Info|Router|Application mode: Interactive
2021-3-16 15:45:23.5|Info|MigrationController|*** Migrating data source=/config/radarr.db;cache size=-20000;datetimekind=Utc;journal mode=Wal;pooling=True;version=3 ***
2021-3-16 15:45:28.4|Info|Bootstrap|Starting Radarr - /app/radarr/bin/Radarr.dll - Version 3.0.2.4552
2021-3-16 15:45:28.6|Info|AppFolderInfo|Data directory is being overridden to [/config]
2021-3-16 15:45:28.6|Info|Router|Application mode: Interactive
2021-3-16 15:45:28.7|Info|MigrationController|*** Migrating data source=/config/radarr.db;cache size=-20000;datetimekind=Utc;journal mode=Wal;pooling=True;version=3 ***
2021-3-16 15:45:28.8|Info|MigrationController|*** Migrating data source=/config/logs.db;cache size=-20000;datetimekind=Utc;journal mode=Wal;pooling=True;version=3 ***
2021-3-16 15:45:29.0|Info|WebHostController|Listening on the following URLs:
2021-3-16 15:45:29.0|Info|WebHostController|  http://*:7878
2021-3-16 15:45:29.2|Info|RadarrBootstrapper|Starting Web Server
2021-3-16 15:45:29.9|Info|CommandExecutor|Starting 2 threads for tasks.
2021-3-16 15:46:00.5|Info|DiskScanService|Scanning disk for 10 Cloverfield Lane
2021-3-16 15:46:00.5|Info|DiskScanService|Completed scanning disk for 10 Cloverfield Lane
```
